addon.py

```diff
@@ -291,19 +291,24 @@ def favourite_tracks():
 
 @plugin.route('/search')
 def search():
-    dialog = xbmcgui.Dialog()
-    fields = ['artist', 'album', 'playlist', 'track']
-    names = ['Artists', 'Albums', 'Playlists', 'Tracks']
-    idx = dialog.select('Search for', names)
-    if idx != -1:
-        field = fields[idx]
-        query = dialog.input('Search')
-        if query:
-            res = wimp.search(field, query)
-            view(res.artists, urls_from_id(artist_view, res.artists), end=False)
-            view(res.albums, urls_from_id(album_view, res.albums), end=False)
-            view(res.playlists, urls_from_id(playlist_view, res.playlists), end=False)
-            track_list(res.tracks)
+	add_directory('Artist', searchtype, _search='artist')
```

Comment:
change `_search` everywhere to `field`. better name and there's no need for underscores